org.lflang/src/org/lflang/generator/ts/TSGenerator.kt

```diff
@@ -383,15 +382,92 @@ class TSGenerator(
         serializer: SupportedSerializers
     ): String {
         return with(PrependOperator) {"""
-        // FIXME: For now assume the data is a Buffer, but this is not checked.
-        // Replace with ProtoBufs or MessagePack.
+        |// FIXME: For now assume the data is a Buffer, but this is not checked.
+        |// Replace with ProtoBufs or MessagePack.
+        |// generateNetworkReceiverBody
         |if (${action.name} !== undefined) {
-        |    ${receivingPort.container.name}.${receivingPort.variable.name} = 
-        |    ${action.name}; // defaults to utf8 encoding
+        |    ${receivingPort.container.name}.${receivingPort.variable.name} = ${action.name}.toString(); // defaults to utf8 encoding
         |}
         """.trimMargin()}
     }
 
+    /**
+     * Generate code for the body of a reaction that handles an output
+     * that is to be sent over the network. This base class throws an exception.
+     * @param sendingPort The output port providing the data to send.
+     * @param receivingPort The ID of the destination port.
+     * @param receivingPortID The ID of the destination port.
+     * @param sendingFed The sending federate.
+     * @param sendingBankIndex The bank index of the sending federate, if it is a bank.
+     * @param sendingChannelIndex The channel index of the sending port, if it is a multiport.
+     * @param receivingFed The destination federate.
+     * @param type The type.
+     * @param isPhysical Indicates whether the connection is physical or not
+     * @param delay The delay value imposed on the connection using after
+     * @throws UnsupportedOperationException If the target does not support this operation.
+     */
+    override fun generateNetworkSenderBody(
+        sendingPort: VarRef,
+        receivingPort: VarRef,
+        receivingPortID: Int,
+        sendingFed: FederateInstance,
+        sendingBankIndex: Int,
+        sendingChannelIndex: Int,
+        receivingFed: FederateInstance,
+        type: InferredType,
+        isPhysical: Boolean,
+        delay: Delay?,
+        serializer: SupportedSerializers
+    ): String {
+        return with(PrependOperator) {"""
+        |// FIXME: For now assume the data is a Buffer, but this is not checked.
+        |// Replace with ProtoBufs or MessagePack.
```

Comment:
Just regarding the comment, the `SupportedSerializers` should be what dictates the serialization. There are both `native` and `proto` serializer options.

Reply:
Ah, I just noticed SupportedSerializers was added here after I started working on the code and got added here when I pulled the recent master to this branch.

I updated this FIXME to use serializer in https://github.com/lf-lang/lingua-franca/pull/538/commits/c58677c8c28bb8b9b785856b0ee61d681d99a8e9.

By the way, this FIXME was here since I just copied and pasted the code used in the legacy TypeScript federated execution example (which doesn't work for now) here without thinking about it too much: https://github.com/lf-lang/reactor-ts/blob/master/src/example/generated/DistributedLogical/Distributed_dsp.ts

Reply:
I think now that the TS target is getting attention, we should keep these TS federated functionalities up-to-date with any changes related to federated execution. I will make a mental note to look here whenever I make a change related to federated execution. Adding tests will also help with this matter. We should also make changes more collaboratively between all of us going forward.

Reply:
Just as a heads-up, I am going to make changes to how and where output control reactions are generated. These changes are needed for a correct functionality. I don't expect these changes to affect the TS target that much, but to err on the safe side, we can have a meeting about it if you are interested.

Reply:
Sure, having a meeting sounds good to me! I'd like to learn the overall communication and control flow between the RTI and federates as well. What would be a good time for a meeting for you? Maybe we could invite others who might be interested in as well? For the upcoming three weeks, I'll be mostly available before 6 PM in Pacific time.